### ISSUE TITLE
Canary returns an empty result when the first decoded token is `<|endoftext|>`

### ISSUE THREAD

**Issue by Mr-Sunglasses** [k2-fsa/sherpa-onnx#3919]
**Version:** 1.13.7 (`917bed9`)
**Model:** `sherpa-onnx-nemo-canary-180m-flash-en-es-de-fr-int8`
**Platform:** macOS 15, Apple Silicon, CPU

### Symptom

`SherpaOnnxGetOfflineStreamResult` returns an empty `text` for ordinary speech. No error is raised, so a caller cannot distinguish it from silence. The decode is suspiciously fast — ~0.13s for 4s of audio, i.e. encoder time only.

### Cause

`offline-recognizer-canary-impl.h`, `DecodeStream`:

```cpp
int32_t max_token_id = GetMaxTokenId(&logits);   // line 72
std::vector<int32_t> tokens = {max_token_id};    // line 79

for (int32_t i = 1; i <= num_tokens; ++i) {
  if (tokens.back() == eos) break;               // line 87 — fires immediately
  ...
}
tokens.pop_back();                               // line 98 — empties the vector
```

When the argmax of the first decoder logits is `<|endoftext|>`, the loop never runs, `pop_back()` empties `tokens`, and the result is `""`. The first generated position is never a legitimate place for end-of-transcript when the audio contains speech.

### Reproduction

The trigger is not a property of the audio you can measure — it is a knife edge. From one 4.0s English recording (16kHz mono) that decodes to `""`:

| input | result |
|---|---|
| the recording as-is | `""` |
| every sample x 1.001 | `"Hey, let's see if this sucks for eight seconds of audio."` |
| every sample x 0.999 | `""` |
| speech shifted +100ms | full sentence |
| speech shifted +300ms | `""` |
| speech shifted +500ms | full sentence |

A 0.1% amplitude change flips it. Level, duration, DC offset, noise floor and language do not separate failures from successes.

The failing file: [`canary-empty-decode-4s.wav`](https://github.com/Mr-Sunglasses/vocamac/raw/sherpa-canary-empty-decode-sample/canary-empty-decode-4s.wav) (4.0s, 16kHz mono). Decoding it with `sherpa-onnx-offline` and the model above reproduces the empty result directly.
